### PR TITLE
Invert Zindex order for children of group controls

### PR DESCRIPTION
**Pull request by lesaltzm**
It looks like #623 changed how ZIndices are applied to controls, so #622 was no longer in effect. This brings back the reordering of zindex for children of group containers.

### CODE REVIEW

File: src/Persistence/Models/Control.cs [hunk 1]
```diff
@@ -94,6 +94,8 @@ public int ZIndex
     [OnDeserialized]
     internal virtual void AfterDeserialize()
     {
+        // Is this dead code now? Looks like it's not being hit anymore after the schema changes?
```
Comment (lesaltzm) [inline src/Persistence/Models/Control.cs:97]: Will clean up this comment, but wanted to check whether this was dead code or not?

Comment (mizrael) [reply src/Persistence/Models/Control.cs:97]: @anpetroc can you chime in? I was trying to debug this method this morning, but it doesn't get called (at least not when running the tests). If that's the case, I'd say we can get rid of it

File: src/Persistence.Tests/_TestData/ValidYaml/ZIndexOrdering/With-list-of-container-controls-with-children.pa.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+- GroupContainer: 
+  Name: Group1
```
Comment (petrochuk) [inline src/Persistence.Tests/_TestData/ValidYaml/ZIndexOrdering/With-list-of-container-controls-with-children.pa.yaml:2]: this one is missing

`Control: Group`

File: src/Persistence/Models/Control.cs (hunk 1 above)
Comment (petrochuk) [reply src/Persistence/Models/Control.cs:97]: It probably is dead code now as we are switching to some new object model hooks

**Review by petrochuk — APPROVED** [review]

**Pull request merged**